shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -668,32 +672,64 @@ public long configureForTextureLayerComposition(
     return textureId;
   }
 
+  /**
+   * Translates an original touch event to have the same locations as the ones that Flutter
+   * calculates (because original + flutter's - original = flutter's).
+   *
+   * @param originalEvent The saved original input event.
+   * @param pointerCoords The coordinates that Flutter thinks the touch is happening at.
+   */
+  private static void translateMotionEvent(
+      MotionEvent originalEvent, PointerCoords[] pointerCoords) {
+    if (pointerCoords.length < 1) {
+      return;
+    }
+
+    float xOffset = pointerCoords[0].x - originalEvent.getX();
+    float yOffset = pointerCoords[0].y - originalEvent.getY();
+
+    originalEvent.offsetLocation(xOffset, yOffset);
+  }
+
   @VisibleForTesting
   public MotionEvent toMotionEvent(
       float density, PlatformViewsChannel.PlatformViewTouch touch, boolean usingVirtualDiplay) {
     MotionEventTracker.MotionEventId motionEventId =
         MotionEventTracker.MotionEventId.from(touch.motionEventId);
     MotionEvent trackedEvent = motionEventTracker.pop(motionEventId);
 
+    // Pointer coordinates in the tracked events are global to FlutterView
+    // The framework converts them to be local to a widget, given that
+    // motion events operate on local coords, we need to replace these in the tracked
+    // event with their local counterparts.
+    // Compute this early so it can be used as input to translateNonVirtualDisplayMotionEvent.
+    PointerCoords[] pointerCoords =
+        parsePointerCoordsList(touch.rawPointerCoords, density)
+            .toArray(new PointerCoords[touch.pointerCount]);
+
     if (!usingVirtualDiplay && trackedEvent != null) {
-      // We have the original event, deliver it as it will pass the verifiable
+      // We have the original event, deliver it after offsetting as it will pass the verifiable
       // input check.
+      translateMotionEvent(trackedEvent, pointerCoords);
+      if (BuildConfig.DEBUG && Build.VERSION.SDK_INT >= API_LEVELS.API_30) {
```

Comment:
FWIW I really dont like when codepaths change between release and debug. 

I think I would prefer more verbose logging and not a crash here. 
That said if there is no action a customer can make that might violate https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#error-messages-should-be-useful

Reply:
Hmm actually, I need to remove this error throwing regardless. I re-discovered something I had forgotten while testing, which is that an individual tap on the screen triggers many motion events, corresponding to (1) on finger down, (a bunch) while you hold, and then (1) on finger up. 

The first and last events are verified, while the intermediate ones aren't (even before we modify them). So we can't throw when detecting an un-verified event.

Reply:
Removed